change lowpass filter CLI names

ctzsnooze opened the pull request:
This PR is intended to give a consistent structure to the names of the lowpass filter-related CLI variables.

At present, the dynamic lowpass value names all start with `dyn_lpf…`, whereas static lowpass value names start with `gyro_lowpass….`  Here are the proposed CLI names on left vs new on right:

Proposed | Original
--- | ---
gyro_lpf1_type | gyro_lowpass_type
gyro_lpf1_static_hz | gyro_lowpass_hz
gyro_lpf1_dyn_min_hz | dyn_lpf_gyro_max_hz
gyro_lpf1_dyn_max_hz | dyn_lpf_gyro_max_hz
gyro_lpf1_dyn_expo | dyn_lpf_gyro_curve_expo
gyro_lpf2_type | gyro_lowpass2_type
gyro_lpf2_static_hz | gyro_lowpass2_hz
dterm_lpf1_type | dterm_lowpass_type
dterm_lpf1_static_hz | dterm_lowpass_hz
dterm_lpf1_dyn_min_hz | dyn_lpf_dterm_max_hz
dterm_lpf1_dyn_max_hz | dyn_lpf_dterm_max_hz
dterm_lpf1_dyn_expo | dyn_lpf_dterm_curve_expo
dterm_lpf2_type | dterm_lowpass2_type
dterm_lpf2_static_hz | dterm_lowpass2_hz

Previously, to list all the dynamic filter settings, we would use `get dyn_g`- but that wouldn't include the static settings.  And vice versa; `get gyro_l`pulls static values, but then we don't get the dynamic settings.  

If the names were consistent, for example as below,  a `get gyro_lpf1` command will return all gyro lowpass1 fields, and `get gyro_l` would return all gyro lowpass fields'.

The downside is that a re-naming like this would mean that filter settings transferred by copy and paste updaters from 4.2 to 4.3 would be lost.  Custom filter settings using old names in the copy and paste would revert to defaults unless they were re-configured manually.  

However, it has always been our position to not support copy and paste upgrades for point version changes.  Also, in 4.3 we are enabling sliders by default, and are providing new, and higher default gyro filter values, because they fly better and work better than the old defaults.

Hence it's actually quite good to make changes of this kind now; it will ensure that our newer defaults, which work best with the slider architecture we are now enabling by default, will get applied for most users even if they copy and paste.

haslinghuis commented on src/main/cli/settings.c:
```diff
@@ -668,9 +668,9 @@ const clivalue_t valueTable[] = {
-    { "dyn_lpf_gyro_min_hz",        VAR_UINT16 | MASTER_VALUE, .config.minmaxUnsigned = { 0, DYN_LPF_FILTER_FREQUENCY_MAX }, PG_GYRO_CONFIG, offsetof(gyroConfig_t, dyn_lpf_gyro_min_hz) },
-    { "dyn_lpf_gyro_max_hz",        VAR_UINT16 | MASTER_VALUE, .config.minmaxUnsigned = { 0, DYN_LPF_FILTER_FREQUENCY_MAX }, PG_GYRO_CONFIG, offsetof(gyroConfig_t, dyn_lpf_gyro_max_hz) },
-    { "dyn_lpf_gyro_curve_expo",    VAR_UINT8  | MASTER_VALUE, .config.minmaxUnsigned = { 0, 10 }, PG_GYRO_CONFIG, offsetof(gyroConfig_t, dyn_lpf_curve_expo) },
+    { "gyro_lpf1_dyn_min_hz",       VAR_UINT16 | MASTER_VALUE, .config.minmaxUnsigned = { 0, DYN_LPF_FILTER_FREQUENCY_MAX }, PG_GYRO_CONFIG, offsetof(gyroConfig_t, dyn_lpf_gyro_min_hz) },
```
Should change `DYN_LPF_FILTER_FREQUENCY_MAX` to `GYRO_LPF_FILTER_FREQUENCY_MAX`.
Same on other places.
Maybe also change variables in firmware too to make it coherent.

ctzsnooze commented:
I have updated variable names within the firmware as follows:

Original | Proposed | Comment
--- | ---
gyro_lpf1_type | gyro_lowpass_type | all gyro lowpass filters now start with `gyro_lpf...`
gyro_lpf1_static_hz | gyro_lowpass_hz | filter type is `static` vs `dyn` for dynamic
gyro_lpf1_dyn_min_hz | dyn_lpf_gyro_min_hz | 
gyro_lpf1_dyn_max_hz | dyn_lpf_gyro_max_hz | 
gyro_lpf1_dyn_expo | dyn_lpf_gyro_curve_expo | 
gyro_lpf2_type | gyro_lowpass2_type | 
gyro_lpf2_static_hz | gyro_lowpass2_hz | always static, but easier to interpret code if we know filter is static
dterm_lpf1_type | dterm_lowpass_type | 
dterm_lpf1_static_hz | dterm_lowpass_hz | 
dterm_lpf1_dyn_min_hz | dyn_lpf_dterm_min_hz | 
dterm_lpf1_dyn_max_hz | dyn_lpf_dterm_max_hz | 
dterm_lpf1_dyn_expo | dyn_lpf_dterm_curve_expo | 
dterm_lpf2_type | dterm_lowpass2_type | 
dterm_lpf2_static_hz | dterm_lowpass2_hz | 
FILTER_LPF1 | FILTER_LOWPASS | used only for gyro / D filters so LPF in name
FILTER_LPF2 | FILTER_LOWPASS2 | 
LPF_MAX_HZ | FILTER_FREQUENCY_MAX | to identify that this particular max hz relates to an `lpf`on gyro or D
DYN_LPF_MAX_HZ | DYN_LPF_FILTER_FREQUENCY_MAX | `FILTER` is redundant, `HZ` indicates that it is a cut-off
GYRO_LPF1_DYN_MIN_HZ | DYN_LPF_GYRO_MIN_HZ_DEFAULT | 
GYRO_LPF1_DYN_MAX_HZ | DYN_LPF_GYRO_MAX_HZ_DEFAULT | 
GYRO_LPF2_HZ_DEFAULT | GYRO_LOWPASS_2_HZ_DEFAULT | 
DTERM_LPF2_HZ_DEFAULT | DTERM_LOWPASS_2_HZ_DEFAULT | 
DYN_LPF_DTERM_MIN_HZ_DEFAULT | DTERM_LPF1_DYN_MIN_HZ_DEFAULT  
DYN_LPF_DTERM_MAX_HZ_DEFAULT | DTERM_LPF1_DYN_MAX_HZ_DEFAULT  
TABLE_GYRO_LPF_TYPE | TABLE_LOWPASS_TYPE | 
TABLE_DTERM_LPF_TYPE | TABLE_DTERM_LOWPASS_TYPE | 
gyro_lowpass1_init_hz | gyro_lowpass_hz | where `gyro_lowpass_hz`was used as a local variable to hold an init value

Notes:
- `lpf`, rather than `lowpass`, is used to provide a searchable phrase that is unique to gyro and D filter configurations
- `lpf1` is used rather than the 'unspecified' `lowpass` when something applies to the first lowpass.
- all gyro filter paraeters start with `gyro...` and all D filters start with `dterm`
- the dynamic or static mode of the filter is indicated with `dyn` or `static` in the variable name
- to avoid a need to change blackbox settings, I haven't changed the names sending data to blackbox.

ctzsnooze commented:
Should be considered after https://github.com/betaflight/betaflight/pull/10856 ; a rebase will, I think, be required.

KarateBrot commented on src/main/sensors/gyro_init.c:
```diff
@@ -223,33 +223,33 @@ static void dynLpfFilterInit()
 void gyroInitFilters(void)
 {
-    uint16_t gyro_lowpass_hz = gyroConfig()->gyro_lowpass_hz;
+    uint16_t gyro_lowpass1_init_hz = gyroConfig()->gyro_lpf1_static_hz;
```
I think it's better to keep the proposed name ruling here as well and rename the local variable 
```cpp
gyro_lowpass1_init_hz
```
to
```cpp
gyro_lpf1_init_hz
```
even though it's just a local variable.

ctzsnooze replied:
thanks.  done.

I'll do the blackbox changes later, don't want to delay this getting merged.

KarateBrot reviewed (approved):
Looks very good. The naming structure makes a lot of sense now.

But we should also take the changes over to the blackbox header at some point, to keep it consistent, imo.

ctzsnooze commented:
Have now changed the blackbox header lines, addressing all feedback to date.  Squashed, tested, ready for merge.

Matching blackbox PR:  https://github.com/betaflight/blackbox-log-viewer/pull/535

mikeller reviewed (commented):
A couple of points here:
- this now needs a rebase;
- with the amount of changes to parameter names that have been done and are proposed here, this means that a lot of forum articles, instruction pages, configuration snippets, and tutorial videos for configuring Betaflight that are out there will stop working for 4.3. I am starting to fear that this will result in a lot of user rejection similar to what happened with 3.5 => 4.0.

ctzsnooze commented:
Thanks, Mike.
Rebased, tested.
Presets, a generally stronger Configurator interface with more effective sliders, more effective CLI searching and auto-completion, and the understanding that 4.3 is a major update, should offset the inconvenience of adapting to newer names.
I agree that a 'copy and paste' updated gains the benefit of 'auto-applying' new defaults (eg higher gyro filters for instance) for fields with CLI name changes, with the downside of 'unexpectedly' losing custom changes.  So many things are changing with PID sliders, feedforward and filtering, that 'copy and paste' updaters, which we always advise against, will need to take special care.

FritzWlan commented:
As one who writes a lot of articles around Betaflight tuning I would not worry about name changes too much. BF 4.3 already renamed and changes the meaning of a lot of parameters around FF, RC Smoothing, Rates and so one. Users finding their way into the CLIncan be considered as advanced. 

What would help to clear up the issue would be a dictionary like function to look up the meaning of each parameter, with a link to an wiki article or website knowledge base article explaining it further.
